torch/_subclasses/meta_utils.py

```diff
@@ -382,7 +383,11 @@ def is_c_of_r(complex_dtype, real_dtype):
                             if t._base.requires_grad == t.requires_grad:
                                 # Easy case, just run the view op
                                 with torch.enable_grad(), maybe_suppress():
-                                    r = base.as_strided(sizes, strides, storage_offset)
+                                    if not t.is_nested:
+                                        r = base.as_strided(sizes, strides, storage_offset)
+                                    else:
+                                        # nested tensor does not support as_strided
+                                        r = t._view_func_unsafe(base)
```

Comment:
TODO: We can probably easily handle the other cases too

Reply:
More generally, I should call view_func_unsafe instead of as_strided for any tensor that does not support as_strided (but that attribute is not binded to python today)

Reply:
Can you explain in the comments here why calling unsafe is OK and why not calling it would not work?

Reply:
> TODO: We can probably easily handle the other cases too

I think we should do this :p or at least error nicely for NT on those branches

Reply:
Updated to actually handle these cases.